samples/HttpClientFactorySample/Program.cs

```diff
@@ -58,6 +66,24 @@ public static void Configure(IServiceCollection services)
             // Build a totally custom policy using any criteria
             .AddPolicyHandler(Policy.Handle<HttpRequestException>().RetryAsync())
 
+            // Use a specific named policy from the registry. Simplest way, policy is cached for the
+            // lifetime of the handler.
+            .AddPolicyHandlerFromRegistry("regular")
+
+            // Run some code to select a policy based on the request
+            .AddPolicyHandler((request) =>
+            {
+                return request.Method == HttpMethod.Get ? timeout : longTimeout;
+            })
+
+            // Run some code to select a policy from the registry based on the request
+            .AddPolicyHandlerFromRegistry((reg, request) =>
+            {
+                return request.Method == HttpMethod.Get ?
+                    reg.Get<IAsyncPolicy<HttpResponseMessage>>("regular") :
+                    reg.Get<IAsyncPolicy<HttpResponseMessage>>("long");
+            })
```

Comment:
This is the best place to start looking. Let's try to nail down what the configuration experience is like and then let the actual implementation take care of itself 😆